go/database/mpt/state.go

```diff
@@ -345,7 +345,7 @@ func (s *MptState) Apply(block uint64, update common.Update) (archiveUpdateHints
 	return hints, err
 }
 
-func (s *MptState) Visit(visitor NodeVisitor) error {
+func (s *MptState) Visit(visitor NodeVisitor, _ bool) error {
```

Comment:
This extra bool (likely the "cutAtAccount") should not be ignored. Actually, it should not be added at all, since the "NodeVisitor" tells the `MptState` what to include and what not by returning visitor responses like "abort", "continue" or "prune".

Instead of updating the this function, a wrapper will be required, customizing the visitor to "prune" when reaching an account, defined by the second flag.